cmd/gateway_server/server/main.go

```diff
@@ -0,0 +1,190 @@
+	if err != nil {
+		return errorToSXGResponse(err), nil
+	}
+
+	signUrl, err := url.Parse(request.SignUrl)
+	if err != nil {
+		return errorToSXGResponse(err), nil
+	}
+
+	var dotStarPattern = ".*"
+	signUrlPattern := util.URLPattern {
+                Domain:    signUrl.Host,
+		QueryRE: &dotStarPattern,
+	}
+	err = util.ValidateSignURLPattern(&signUrlPattern)
+	if err != nil {
+		return errorToSXGResponse(err), nil
+	}
+
+	fetchUrlPattern := util.URLPattern {
+		Scheme:    []string{"http"},
+		Domain:    fmt.Sprintf("localhost:%d", *publisherServerPort),
+		ErrorOnStatefulHeaders: false,
+		QueryRE: &dotStarPattern,
+                SamePath:  new(bool),
+	}
+	*fetchUrlPattern.SamePath = false
+	err = util.ValidateFetchURLPattern(&fetchUrlPattern)
+	if err != nil {
+		return errorToSXGResponse(err), nil
+	}
+
+	urlSets := []util.URLSet{
+		{
+			Sign: &signUrlPattern,
+			Fetch: &fetchUrlPattern,
+		},
+	}
+
+	packager, err := signer.New(certs[0], privateKey, urlSets, rtvCache, shouldPackage, signUrl, false)
+
+	if err != nil {
+		return errorToSXGResponse(err), nil
+	}
+
+	if packager == nil {
+		return errorToSXGResponse(err), nil
+	}
+
+	baseUrl, err := url.Parse(fmt.Sprintf("http://localhost:%d", *port))
+	if err != nil {
+		return errorToSXGResponse(err), nil
+	}
+	q := baseUrl.Query()
+	q.Set("fetch", request.FetchUrl)
+	q.Set("sign", request.SignUrl)
+	baseUrl.RawQuery = q.Encode()
+
+	httpreq, err := http.NewRequest("GET", baseUrl.String(), nil)
+	httpresp := httptest.NewRecorder()
+	packager.ServeHTTP(httpresp, httpreq, httprouter.Params{})
+	if httpresp.Code != 200 {
+		// TODO(amaltas): Add counter.
+		return &pb.SXGResponse{
+			Error: true,
+			ErrorDescription: "Packager error.",
+		}, nil
+	}
+
+	// Creates cbor data.
+	ocspDer := []byte("ocsp")
+	var sctList []byte
+	certChain, err := certurl.NewCertChain(certs, ocspDer, sctList)
+	if err != nil {
+		return errorToSXGResponse(err), nil
+	}
+
+	buf := &bytes.Buffer{}
+	err = certChain.Write(buf)
+	if err != nil {
+		return errorToSXGResponse(err), nil
+	}
+
+	// Record http headers from the packager.
+	http_headers := make(map[string]string)
+	for header_key, header_value := range httpresp.Header() {
+		// Ignores multiple header values.
+		http_headers[strings.ToLower(header_key)] =
+		string(header_value[0])
+	}
+
+	return &pb.SXGResponse{
+		Sxg: httpresp.Body.Bytes(),
+	        Cbor: buf.Bytes(),
+	        HttpHeaders: http_headers}, nil
+}
+
+func main() {
+	flag.Parse()
+	keyPem, err := ioutil.ReadFile(*privateKey)
+	if err != nil {
+		log.Fatalf("Error reading private key file.")
+	}
+	key, err := util.ParsePrivateKey(keyPem)
+	if (err != nil) {
+		log.Fatalf("Error reading parsed private key string.")
+	}
+	if key == nil {
+		log.Fatalf("Key is nil.")
+	}
+
+	if *port == -1 {
+		log.Fatalf("Set flag -port")
+	}
+	if *publisherServerPort == -1 {
+		log.Fatalf("Set falg -publisher-server-port")
+	}
+
+	lis, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", *port))
```

Comment:
I'd prefer s/lis/listener/ personally, but it's up to you. (I know golang convention is to prefer short variable names, but I disagree.)